chore(node): Add preinstall check for node version

jeff-phillips-18 opened the pull request:
**What**:
We require node version 8. Failure raised is not intuitive. This make it obvious that you need node 8.

priley86 commented:
@jeff-phillips-18 maybe we can try the `postinstall` hook [instead](https://www.npmjs.com/package/nvmrc-check)? I believe Travis should honor the node version present in `.travis.yml`.

jeff-phillips-18 commented:
Changed to use `postinstall` hook. Also added a section to the readme file to help contributors thru some of the install issues when starting out with Node < 8.
